### DISCUSSION TITLE
Batch Graphql @Source

### DISCUSSION THREAD

**Discussion by uPagge** [quarkusio/quarkus#29330] [Q&A] [answered]
I tried to use batch queries in GraphQL according to [this guide](https://quarkus.io/blog/experimental_graphql/), but I ran into an unpleasant problem.

```
2022-11-17 14:00:00,240 ERROR [io.sma.graphql] (vert.x-eventloop-thread-2) SRGQL012000: Data Fetching Error: org.dataloader.impl.DataLoaderAssertionException: The size of the promised values MUST be the same size as the key list
	at org.dataloader.impl.Assertions.assertState(Assertions.java:12)
	at org.dataloader.DataLoaderHelper.assertResultSize(DataLoaderHelper.java:282)
	at org.dataloader.DataLoaderHelper.lambda$null$7(DataLoaderHelper.java:382)
```

The problem is that several different entities can have the same nested entities. When these nested entities are requested, their collection is smaller than the collection in the @Source method parameters. And in my opinion, this is normal, because there may not be a nested entity at all, or there may be fewer of them due to duplication.

But in this case I get an error. If you duplicate the entity with your hands in the collection, then everything starts working, but this is some kind of crutch

My request looks something like this
```java
public Uni<List<PersonData>> getPhysicalPerson(@Source List<AppealOutput> appeals) { // list.size() == 10
        return Multi.createFrom().iterable(appeals)
                .onItem().transform(appeal -> appeal.getInitiator().getId())
                .collect().asList()
                .onItem().transformToUni(physicalPersonClient::getAllById) // example list.size() == 5, error
    }
```

<img width="1249" alt="screen 7" src="https://user-images.githubusercontent.com/40397740/202433073-748a9222-6d2a-4732-8578-f83f08a42d64.png">

[a comment by an automated account is not shown]

**Reply by jmartisk** [reply]
I think that's just how it works in the graphql-java library - the resulting list must be the same size, so that it can match `AppealOutput` with the corresponding `PersonData`, so yes, if multiple `AppealOutput` map to the same person, you'll have to duplicate the person. Otherwise the library won't know how to map these entities together.

**Reply by uPagge** [reply]
At the moment, this is exactly the path I have gone. But it hurts so strange and cumbersome looks
